compiler/src/dmd/dmodule.d

```diff
@@ -351,6 +351,7 @@ extern (C++) final class Module : Package
     extern (C++) __gshared AggregateDeclaration moduleinfo;
 
     const(char)[] arg;           // original argument name
+    bool specifiedOnCmdLine = true;    // Was the module explicitly mentioned on the compiler invocation.
```

Comment:
That's what `isRoot` is for.

Reply:
But that isn't what isRoot actually does

Reply:
A module imported by -i is root

Reply:
> A module imported by -i is root

When using `-i` you _are_ mentioning it on the command line though.

A module that imports itself (or has another module import it) would surely have this set to false as well if your only metric is `Compiler::onImport`

Reply:
You aren't mentioning it in any useful sense when it comes to running unittests.

Reply:
The exact details can be hashed out but fundamentally the idea is to get on the road towards `dmd -unittest -i -run blah.d" not running a bunch of extra tests you don't care about